### PR TITLE
23 Resql endpoints for chat metrics

### PR DESCRIPTION
**Pull request by gdrui**
Metric for getting the percentage of correctly understood messages is based on a hypothetical 'not-confident' event that would be marked by the chatbot when responding to a message the bot did not understand.

### CODE REVIEW

**Review by turnerrainer — COMMENTED** [review]
Please add newlines at end of files.

File: DSL/Resql/get-chat-count-total.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+SELECT COUNT(distinct base_id)
```
Comment (baha-a) [inline DSL/Resql/get-chat-count-total.sql:1]: Please change `distinct` to all upper case.

File: DSL/Resql/get-avg-waiting-time.sql [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        ) AS prev_support_id,
+        lag(updated) over (
+            PARTITION by base_id
+            ORDER BY updated
+        ) AS prev_updated
+    FROM chat
+    WHERE created BETWEEN :start::timestamptz AND:
+END::timestamptz
+)
+SELECT avg(
```
Comment (baha-a) [inline DSL/Resql/get-avg-waiting-time.sql:22]: Please change `avg` to all upper case.

**Review by baha-a — CHANGES_REQUESTED** [review]

File: DSL/Resql/get-avg-messages.sql [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    FROM chat
+    WHERE created BETWEEN :start::timestamptz AND :end::timestamptz
+        AND EXISTS (
+            SELECT 1
+            FROM message
+            WHERE message.chat_base_id = chat.base_id
+                AND message.author_role = 'end-user'
+        )
+)
+SELECT AVG(num_messages)
```
[the remark this answers is not in the document]
Comment (gdrui) [reply DSL/Resql/get-avg-messages.sql:12]: Very good point, I'll wrap these with COALESCE, its shorter

File: DSL/Resql/get-pct-correctly-understood.sql [hunk 4]
```diff
@@ -0,0 +1,16 @@
+WITH chats AS (
+    SELECT DISTINCT base_id
+    FROM chat
+    WHERE created BETWEEN :start::timestamptz AND :end::timestamptz
+)
+SELECT ROUND(
+        (
+            1 - COUNT(*) filter (
+                WHERE event = 'not-confident'
+            )::numeric / COUNT(*)::numeric
```
[the remark this answers is not in the document]
Comment (gdrui) [reply DSL/Resql/get-pct-correctly-understood.sql]: Good point

**Review by baha-a — APPROVED** [review]

**Pull request merged**